### ISSUE TITLE
Customize styles for each of the axises items.

### ISSUE THREAD

**Issue by fqueiruga** [JesperLekland/react-native-svg-charts#76]
In our project we need to customize the axis item styles indivitually on a chart. AFAIK, the axis only allows setting styles for all the items at once.

What I'm trying to achieve (:
<img width="305" alt="multicoloraxis" src="https://user-images.githubusercontent.com/5738588/36948489-e445832c-1fdb-11e8-9b0b-70be78b121a4.png">

I think a nice solution for this would be making the `svg` prop on the XAxis and YAxis components to also accept a function. This function would be called with the `value` and `index` arguments, same as the `formatLabel` prop, and would return an object with the styles. This approach wouldn't break backwards compatibility.

Sample code:
```javascript
    <XAxis
            data={data}
            formatLabel={this.formatDayLabel}
            svg={(value, index) => ({
              fontSize: 10,
              fontWeight: index < data.length - 1 ? 'normal' : 'bold',
              fill: index < data.length - 1 ? Color.AXIS_GREY : Color.AXIS_BLUE,
            })}
            scale={scale.scaleBand}
            spacing={0.5}
            />
```

The implementation could be something like this (on the `XAxis` component:
```javascript
    <SVGText
                  textAnchor={'middle'}
                  originX={x(value)}
                  alignmentBaseline={'hanging'}
                  {...(svg instanceof Function ? svg(value, index) : svg)}
                  key={index}
                  x={x(value)}
                >
                  {formatLabel(value, index)}
                </SVGText>
```

I understand this approach could not be ideal and coherent if it isn't used in all the other components. Another solution would be creating another prop without altering the `svg` prop.

I'd love to contribute and implement this solution and expanding the docs if you like this approach.

**Comment by JesperLekland** [comment]
Hi again and thank you once more for your support :) 

I think the best approach would be to align it with the other charts API's, e.g letting the user add an `svg` property to each entry in `data` and having that override the "global" svg for the Axis. See for example what's done with Bar/PieChart when you override `svg` or `arc`. This wouldn't break the existing API either as it already allows for complex arrays to be passed to the axes

**Comment by fqueiruga** [comment]
Thank you for replying so quickly! Lovely library.

I agree, I was just taking a look at the 4.0.0 release changes and this approach makes more sense. I'd love to work on this tomorrow.

What would a PR need to be accepted, other than changing the code, the prop types and the documentation?

**Comment by JesperLekland** [comment]
Fantastic! :D 

The documentation in general around the new, more powerful, data prop (with complex objects) could use some clarification in order for people to better understand its power. 

In addition to what you've mentioned you should make sure to implement a storybook "story" with an example. In addition to that you should also add said example to [this](https://github.com/JesperLekland/react-native-svg-charts-examples) repository 🥇 

I'll be happy to help with any questions along the way

**Comment by fqueiruga** [comment]
Perfect! I'll give it a try and open a MR

**Comment by JesperLekland** [comment]
Available in 4.1.0
